### DISCUSSION TITLE
Replace MQTT Forwarder with Subscribe Handler (mProxy) of MQTT Adapter

### DISCUSSION THREAD

**Discussion by arvindh123** [absmach/magistrala#2221] [Ideas]
Instead of using a separate MQTT Forwarder that subscribes to all topics from the `Message Bus / Broker` and forwards publishes to MQTT, we can utilize a subscribe handler that subscribes to specific topics on the `Message Bus / Broker` based on the interests of the MQTT client.

Impacts that are not determinable as Pros or Cons by me:
- Each MQTT subscriber will have one NATS client connection on interested topic. 

Pros: 
- This will prevent publishing of unwanted topics to MQTT Broker.

**Comment by dborovcanin** [comment]
It also makes sense to make the adapter configurable, so that we can make this "cross-protocol" communication optional; i.e. we have a flag so we decide whether to make a NATS subscription or not. We also need to be careful so that we do avoid duplicate messages (i.e. all MQTT messages will end in NATS, too, so if we have both MQTT and NATS subscriptions, we need to take only non-MQTT messages from NATS).
